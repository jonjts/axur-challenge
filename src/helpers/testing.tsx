import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
