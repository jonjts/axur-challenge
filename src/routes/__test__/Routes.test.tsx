import { matchSnapshotWithProviders, renderWithProviders } from '../../helpers/testing';

import Routes from '../index';

describe('Routes', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<Routes />);
  });

  test('renders correctly', () => {
    renderWithProviders(<Routes />);
  });
});
