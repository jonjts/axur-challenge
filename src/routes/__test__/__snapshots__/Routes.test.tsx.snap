// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes match with snapshot 1`] = `
Array [
  <header
    className="sc-gsTCUz iUNJdd"
  >
    <a
      href="/"
      title="Ir para home"
    >
      <img
        alt="Logo do app: desafio Axur"
        className="sc-dlfnbm ebWWjJ"
        src="brand.svg"
      />
    </a>
  </header>,
  <div
    className="sc-hKgILt euAZNc"
  >
    <h1>
      Home
    </h1>
    <div
      className="sc-kEjbxe irossh grid"
    >
      <div>
        <div
          className="sc-gKsewC gzXdMk bounce"
          onClick={[Function]}
        >
          <div
            className="sc-iqHYGH ipQvOR"
          >
            <h3>
              Nova Inspeção
            </h3>
            <p>
              Adicione uma nova keyword para iniciar o processo de web crawling.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          className="sc-gKsewC gzXdMk bounce"
          onClick={[Function]}
        >
          <div
            className="sc-iqHYGH ipQvOR"
          >
            <h3>
              Consultar Inspeção
            </h3>
            <p>
              Informe o id de uma inspeção já cadastrada para verificar seus resultados.
            </p>
          </div>
        </div>
      </div>
      <div
        className="sc-fubCfw cqjzZl"
      >
        <h2>
          Inspeções Recentes
        </h2>
        <div
          className="sc-pFZIQ iaPyDd grid"
        />
      </div>
    </div>
  </div>,
]
`;
